connectors/protocol/connectors.py

```diff
@@ -784,11 +784,17 @@ async def prepare(self, config, sources):
             doc["service_type"] = configured_service_type
             self.log_debug(f"Populated service type {configured_service_type}")
 
+        simple_config = source_klass.get_simple_configuration()
         if self.configuration.is_empty():
             # sets the defaults and the flag to NEEDS_CONFIGURATION
-            doc["configuration"] = source_klass.get_simple_configuration()
+            doc["configuration"] = simple_config
             doc["status"] = Status.NEEDS_CONFIGURATION.value
             self.log_debug("Populated configuration")
+        elif simple_config.keys() - self.configuration.to_dict().keys():
```

Comment:
We repeat the expression `simple_config.keys() - self.configuration.to_dict().keys()` twice

Reply:
Yep. I guess I can precompute it, though it wouldn't be used in the first `if`. Ugh. :P